packages/react-core/src/demos/PasswordGenerator.md

```diff
@@ -0,0 +1,91 @@
+---
+id: Password generator
+section: demos
+---
+
+import RedoIcon from '@patternfly/react-icons/dist/esm/icons/redo-icon';
+import EyeIcon from '@patternfly/react-icons/dist/esm/icons/eye-icon';
+import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
+
+## Examples
+
+### Provide a generated password
+
+```js
+import React from 'react';
+import {
+  Form,
+  FormGroup,
+  InputGroup,
+  TextInput,
+  Button,
+  Popover,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  MenuItemAction
+} from '@patternfly/react-core';
+import RedoIcon from '@patternfly/react-icons/dist/esm/icons/redo-icon';
+import EyeIcon from '@patternfly/react-icons/dist/esm/icons/eye-icon';
+import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
+
+PasswordGenerator = () => (
+  <Form>
+    <FormGroup
+      label="Password"
+      isRequired
+      fieldId="password-generator-demo--initial-password"
+      labelIcon={
+        <Popover bodyContent={<div>Generates a password</div>}>
+          <button
+            type="button"
+            aria-label="More info"
+            onClick={e => e.preventDefault()}
+            className="pf-c-form__group-label-help"
+          >
+            <HelpIcon noVerticalAlign />
+          </button>
+        </Popover>
+      }
+    >
+      <InputGroup>
+        <TextInput
+          isRequired
+          type="password"
+          id="password-generator-demo--initial-password"
+          name="password-generator-demo--initial-password"
+          aria-label="Password input"
+          placeholder="Password"
+          iconVariant="fa-eye"
+        />
+        <Button variant="control" aria-label="search button for search input">
```

Comment:
This should be sentence case and toggle between show/hide password. You can reference the login page show/hide password demo for that - https://www.patternfly.org/v4/components/login-page/react/showhide-password/

```suggestion
        <Button variant="control" aria-label="Show password">
```